src/provisioning/provisioner.ts

```diff
@@ -0,0 +1,242 @@
+        return res.send(this.connMan.enabledForProvisioning);
+    }
+
+    private getConnections(req: Request<{roomId: string}>, res: Response<GetConnectionsResponseItem[]>) {
+        const connections = this.connMan.getAllConnectionsForRoom(req.params.roomId);
+        const details = connections.map(c => c.getProvisionerDetails?.()).filter(c => !!c) as GetConnectionsResponseItem[];
+        return res.send(details);
+    }
+
+    private getConnection(req: Request<{roomId: string, connectionId: string}>, res: Response<GetConnectionsResponseItem>) {
+        const connection = this.connMan.getConnectionById(req.params.roomId, req.params.connectionId);
+        if (!connection) {
+            throw new ApiError("Connection does not exist", ErrCode.NotFound);
+        }
+        if (!connection.getProvisionerDetails)  {
+            throw new ApiError("Connection type does not support updates", ErrCode.UnsupportedOperation);
+        }
+        return res.send(connection.getProvisionerDetails());
+    }
+
+    private async putConnection(req: Request<{roomId: string, type: string}, unknown, Record<string, unknown>, {userId: string}>, res: Response, next: NextFunction) {
+        // Need to figure out which connections are available
+        try {
+            if (!req.body || typeof req.body !== "object") {
+                throw new ApiError("A JSON body must be provided", ErrCode.BadValue);
+            }
+            const eventId = await this.connMan.provisionConnection(req.params.roomId, req.query.userId, req.params.type, req.body);
+            return res.status(202).send({eventId});
+        } catch (ex) {
+            log.warn(`Failed to create connection for ${req.params.roomId}`, ex);
+            return next(ex);
+        }
+    }
+
+    private async patchConnection(req: Request<{roomId: string, connectionId: string}, unknown, Record<string, unknown>, {userId: string}>, res: Response<GetConnectionsResponseItem>, next: NextFunction) {
+        const connection = this.connMan.getConnectionById(req.params.roomId, req.params.connectionId);
+        if (!connection) {
+            return next(new ApiError("Connection does not exist", ErrCode.NotFound));
+        }
+        if (!connection.provisionerUpdateConfig || !connection.getProvisionerDetails)  {
+            return next(new ApiError("Connection type does not support updates", ErrCode.UnsupportedOperation));
+        }
+        try {
+            await connection.provisionerUpdateConfig(req.query.userId, req.body);
+        } catch (ex) {
+            next(ex);
+        }
+        res.send(connection.getProvisionerDetails());
+    }
+
+    private async deleteConnection(req: Request<{roomId: string, connectionId: string}>, res: Response<{ok: true}>, next: NextFunction) {
+        const connection = this.connMan.getConnectionById(req.params.roomId, req.params.connectionId);
```

Comment:
I haven't tried the PATCH or GET endpoints, but this DELETE one doesn't seem to ever find the connection. Tried calling it with `hookshot.getConnections(roomId)[0].id` and it still failed.

I suspect it's a connection ID mismatch or something.

Reply:
on jira btw - probably the same for github?

Reply:
I think fixed.